### PR TITLE
Align UserCountGraph properly.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #2708.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss [hunk 1]
```diff
@@ -46,6 +46,17 @@
 		}
 	}
 
+	.googlesitekit-widget--analyticsAllTraffic__user-count-chart {
+
+		@media (min-width: $bp-desktop) {
+			margin-top: 1.4vw;
```
Comment (felixarntz) [inline assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss:52]: I don't think we should use `vw` here since it is dynamic on the viewport width. The margin shouldn't need to be different though based on how many pixels exactly your viewport is wide. I tried using `13px` here, which brings makes the two more aligned.

File: assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss [hunk 2]
```diff
@@ -46,6 +46,17 @@
...
+		}
+
+		@media (min-width: $bp-xlarge) {
+			margin-top: 0;
+		}
```
Comment (felixarntz) [inline assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss:57]: This is okay, but for this viewport (at least on my end), the pie chart is still slightly higher up, so they're not really aligned. I think you need to add `margin-top` to the `.googlesitekit-widget--analyticsAllTraffic__dimensions-chart` element for this viewport - `2px` did it for me. Let me know if that works on your end as well.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@tofumatt Nit-pick, but let's try to be pixel-perfect here - that might be the only way to get the alignment correctly. Unless there are some implications here that differ between our browsers/viewports, they are still not exactly aligned on my end with this.

Please also review the feedback I left above.

**Comment by tofumatt** [comment]
Oddly, seems the "absolute" values passed to charts still account for viewport size, etc.—it's actually what the adjacent pie chart was using. I switched this component over to using absolute values and it seems to work much better:

**Comment by tofumatt** [comment]
Regarding the date cutoff: from some searching on the web this seems to be a common issue with Google Charts as a whole, and difficult to work around. #2699 could solve it by allowing for "automatic" tick marks, but there's no obvious setting to control date cutoff.

Strange that it's cutting off for you but not me too. I don't think we can easily fix that issue though.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserCountGraph.js [hunk 3]
```diff
@@ -100,12 +107,15 @@ UserCountGraph.chartOptions = {
 		startup: true,
 	},
 	curveType: 'function',
-	height: 340,
+	height: 410,
```
Comment (felixarntz) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserCountGraph.js:110]: There's now a lot of extra whitespace at the bottom of the widget above the "Source" link, which looks odd. Likely due to increasing the height here? Can we achieve the vertical alignment without changing the overall chart height - why was this necessary?

See screenshot of extra whitespace:
Screen Shot 2021-01-28 at 3 36 11 PM

**Review by felixarntz — CHANGES_REQUESTED** [review]
@tofumatt This looks great regarding the issue, the dates now show in their full version too. However, the solution now introduces another problem, unnecessary whitespace between the main widget UI and the source link. Could you check that?

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserCountGraph.js (hunk 3 above)
Comment (tofumatt) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserCountGraph.js:110]: This is as tight as I can get the margins without changing the grid/widget margins on things:

Screenshot 2021-01-29 at 13 32 24

The margins are accounting for most of the whitespace:

Screenshot 2021-01-29 at 13 45 02
Screenshot 2021-01-29 at 13 44 53

But I think my reductions in the margins is an improvement 🙂

**Comment by tofumatt** [comment]
Sure thing! Should be all sorted now 🙂

**Review by felixarntz — APPROVED** [review]
@tofumatt Awesome!

I've pushed one more commit here (https://github.com/google/site-kit-wp/pull/2712/commits/32e2be50d6258fed36aedbae8ca85a9234b16889), which adjusts the height of the `PreviewBlock`s in `TotalUserCount` and `UserCountGraph`. Before, they were not matching the height of the fully loaded UI, resulting in ugly content jumps when selecting another pie slice during loading state.

With the values I specified now, it should result in the widget not changing its height in these loading states.

cc @aaemnnosttv
